Sources/NIO/System.swift

```diff
@@ -625,8 +627,26 @@ internal enum KQueue {
     @inline(never)
     @discardableResult
     public static func kevent(kq: CInt, changelist: UnsafePointer<kevent>?, nchanges: CInt, eventlist: UnsafeMutablePointer<kevent>?, nevents: CInt, timeout: UnsafePointer<Darwin.timespec>?) throws -> CInt {
+        var timeoutClamped = timeout
+        // Note that a nil timeout waits indefinitely
+        if let ts = timeout {
+            // Assert to check that timeout conforms to the timespec definition in the C11 standard (ISO/IEC 9899:2011)
+            assert(ts.pointee.tv_nsec >= 0 || ts.pointee.tv_nsec < 1_000_000_000, "kevent timeout value is malformed (0 <= tv_nsec < 1_000_000_000)")
+            // Validate that the timeout parameter passed to kevent does not exceed the limits of the Darwin kernel
+            // interval timer (note that - whilst unlikely - these values could change)
+            if ts.pointee.tv_sec < 0  || ts.pointee.tv_sec > sysIntervalTimerMaxSec {
```

Comment:
Same note here about swapping to `contains` as well as moving the check out of the wrapper. The conditional is right here though.